http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -505,12 +513,217 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       _ <- publisherQueue.offer(message)
     } yield ()
 
+  override def listAppV2(userInfo: UserInfo, workspaceId: WorkspaceId, params: Map[String, String])(implicit
+    as: Ask[F, AppContext]
+  ): F[Vector[ListAppResponse]] = for {
+    paramMap <- F.fromEither(processListParameters(params))
+    allClusters <- KubernetesServiceDbQueries
+      .listAppsByWorkspaceId(Some(workspaceId), paramMap._1, paramMap._2)
+      .transaction
+
+    res = allClusters
+      .flatMap(c => ListAppResponse.fromCluster(c, Config.proxyConfig.proxyUrlBase, paramMap._3))
+      .toVector
+    // TODO: Figure out permissions
```

Comment:
this permission part should be mostly the same as v1 implementation. might make sense to have a helper function to share the code